orion/primitives/timeseries_postprocessing.py

```diff
@@ -0,0 +1,191 @@
+import operator
+import numpy as np
+import pandas as pd
+
+def extract_dimension(X, dim=None):
+    """
+    The function checks if the dataset being used is valid i.e has a length greater than 0 and contains the dimension required
+    Args:
+        X (ndarray): 
+            N-dimensional value sequence to iterate over
+        dim (int):
+            Integer indicating the dimension number for a multi-dimensional dataset
+    Returns:
+        ndarray:
+            Returns an nd array that contains a dataset with 2 columns ['timestamp', 'value']
+
+    """
+    
+    if (len(X) == 0):
+        return []
+    
+    columns = X.columns.values
+    
+    if 'timestamp' not in columns:
+        X['timestamp'] = X.index.values
+    
+    if dim != None:
+        if dim in columns:
+            X['value'] = X[dim]
+            X = pd.DataFrame().assign(timestamp=X['timestamp'], value=X[dim])
+    
+    if 'value' not in X.columns.values:
+        return []          
+    
+    
+    return X[['timestamp', 'value']]
+
+
+
+def rolling_std_thres(X, thres, op = ">", window_size=5):
```

Comment:
can you change the name of the method to `rolling_std_threshold` to make it clearer?